core/api/rpc/wsc.cpp

```diff
@@ -185,4 +187,22 @@ namespace fc::api::rpc {
       }
     }
   }
+
+  void Client::reconnect(int counter, std::chrono::milliseconds wait) {
+    if(reconnecting.exchange(true)) return;
+    logger_->info("Starting reconnect to {}:{}", client_data.host, client_data.port);
+    for(int i = 0; i < counter; i++){
+      std::this_thread::sleep_for(wait*(i+1));
+      auto res = connect(client_data.host,
+                         client_data.port,
+                         client_data.target,
+                         client_data.token);
```

Comment:
Here socket would be reused. You can not do it this [way](https://github.com/boostorg/beast/issues/1778)